packages/dmn-js-shared/src/base/Manager.js

```diff
@@ -316,13 +331,32 @@ export default class Manager {
   }
 
   /**
-   * Open diagram element.
+   * @typedef {Object} OpenResult
+   *
+   * @property {Array<string>} warnings - Warnings occured during opening
+   */
+
+  /**
+    * @typedef {Error} OpenError
+    *
+    * @property {Array<string>} warnings - Warnings occured during opening
+    */
+
+  /**
+   * Open diagram view.
    *
-   * @param  {ModdleElement}   element
-   * @param  {Function} [done]
+   * @param  {View} view
+   * @returns {Promise} Resolves with {OpenResult} when successful
+   * or rejects with {OpenError}
    */
-  open(view, done=noop) {
-    this._switchView(view, done);
+  open(view) {
```

Comment:
This looks a bit messy. Maybe we can make the wrapping earlier? Something like (didn't try it out)

```js
open = wrapForCompatibility((view) => {
  // ...
})
```

Reply:
We could do that, but then we would introduce arrow functions (which we don't have yet). Therefore I stepped away from it.

But what I will do is bind the context via `apply` to get away from the `self = this`

```
    var open = wrapForCompatibility(function open(view) {
      return this._switchView(view);
    });

    return open.apply(this, arguments);
```

Reply:
How about this?

```diff
diff --git a/packages/dmn-js-shared/src/base/Manager.js b/packages/dmn-js-shared/src/base/Manager.js
index 79bc8641..4c1eb260 100644
--- a/packages/dmn-js-shared/src/base/Manager.js
+++ b/packages/dmn-js-shared/src/base/Manager.js
@@ -63,6 +63,9 @@ export default class Manager {
     this._viewers = {};
 
     this._init(options);
+
+    // keep support for callbacks
+    this.open = wrapForCompatibility(this.open.bind(this));
   }
 
   /**
@@ -347,11 +350,7 @@ export default class Manager {
    * or rejects with {OpenError}
    */
   open(view) {
-    var open = wrapForCompatibility(function open(view) {
-      return this._switchView(view);
-    });
-
-    return open.apply(this, arguments);
+    return this._switchView(view);
   }
 
   _setDefinitions(definitions) {
```

Reply:
So we could just wrap it in the constructor.

Reply:
I like @barmac 's proposal. This allows us to keep the logic unchanged, and then to have a seperate section where we do the compatibility wrapping.

I will adjust accordingly.

Reply:
We also use this pattern in [other places](https://github.com/camunda/camunda-modeler/blob/develop/client/src/app/App.js#L138).